File: Week 4/PRG_레벨1_문자열 내 마음대로 저장하기/hyungjoon_solved.java
import java.util.*;

class Solution {
    public String[] solution(String[] strings, int n) {
        String[] answer = new String[strings.length];
        
        List<StringIndex> temp = new ArrayList<>();
        
        // for문 돌면서 list 형성
        for(String str : strings) {
            temp.add(new StringIndex(str.charAt(n), str));
        }
        
        // n번 문자 기준으로 1차 정렬 후, 문자열 기준으로 2차 정렬
        temp.sort(Comparator.comparing(StringIndex::getC).thenComparing(StringIndex::getString));
        
        // 정답에 넣어주기
        for(int i = 0; i < temp.size(); i++){
            answer[i] = temp.get(i).getString();
        }
        
        return answer;
    }
}

// [문자, 문자열] 형태로 이루어진 클래스
class StringIndex {
    private char c;
    private String str;
    
    public StringIndex(char c, String str) {
        this.c = c;
        this.str = str;
    }
    
    public char getC() {
        return c;
    }
    
    public String getString() {
        return str;
    }
    
}

File: Week 4/PRG_레벨1_문자열 내 마음대로 저장하기/hyungjoon_solved.py
'''
문제 : 문자열 내 마음대로 저장하기
난이도 : 레벨 1
소요 시간 : 22분
'''
def solution(strings, n):
    answer = []

    temp = []
    # [문자, 문자열] 로 구성된 2차원 배열을 만들고, 0번-1번 우선순위로 정렬 시킨다.
    for index, i in enumerate(strings):
        temp.append([i[n], i, index])
    temp.sort(key=lambda x:(x[0], x[1]))

    # 이렇게 생성된 2차원 배열을 for문 돌면서 answer에 넣어주기
    for i in temp:
        answer.append(i[1])

    return answer